src/main/scala/com/cognite/sdk/scala/v1/resources/WellDataLayer.scala

```diff
@@ -0,0 +1,253 @@
+  import WellDataLayer._
+
+  def create(items: Seq[WellboreSource]): F[Seq[Wellbore]] = {
+    val body = WellboreSourceItems(items)
+    requestSession.post[Seq[Wellbore], WellboreItems, WellboreSourceItems](
+      body,
+      uri"${requestSession.baseUrl}/wdl/wellbores",
+      value => value.items
+    )
+  }
+
+  def setMergeRules(rules: WellboreMergeRules): F[Unit] =
+    requestSession.post[Unit, EmptyObj, WellboreMergeRules](
+      rules,
+      uri"${requestSession.baseUrl}/wdl/wellbores/mergerules",
+      discard
+    )
+}
+
+object WellDataLayer {
+  implicit val sourceEncoder: Encoder[Source] = deriveEncoder
+  implicit val sourceDecoder: Decoder[Source] = deriveDecoder
+
+  implicit val deleteSourcesEncoder: Encoder[DeleteSources] = deriveEncoder
+  implicit val sourceItemsDecoder: Decoder[SourceItems] = deriveDecoder
+  implicit val sourceItemsEncoder: Encoder[SourceItems] = deriveEncoder
+
+  implicit val assetSourceEncoder: Encoder[AssetSource] = deriveEncoder
+  implicit val assetSourceDecoder: Decoder[AssetSource] = deriveDecoder
+  implicit val deleteWellsEncoder: Encoder[DeleteWells] = deriveEncoder
+
+  implicit val distanceEncoder: Encoder[Distance] = deriveEncoder
+  implicit val distanceDecoder: Decoder[Distance] = deriveDecoder
+  implicit val wellheadDecoder: Decoder[Wellhead] = deriveDecoder
+  implicit val wellheadEncoder: Encoder[Wellhead] = deriveEncoder
+  implicit val datumDecoder: Decoder[Datum] = deriveDecoder
+  implicit val datumEncoder: Encoder[Datum] = deriveEncoder
+
+  implicit val wellboreDecoder: Decoder[Wellbore] = deriveDecoder
+
+  implicit val wellDecoder: Decoder[Well] = deriveDecoder
+
+  implicit val wellSourceEncoder: Encoder[WellSource] = deriveEncoder
+  implicit val wellSourceItemsEncoder: Encoder[WellSourceItems] = deriveEncoder
+  implicit val wellItemsDecoder: Decoder[WellItems] = deriveDecoder
+  implicit val wellMergeRulesEncoder: Encoder[WellMergeRules] = deriveEncoder
+
+  implicit val wellboreSourceEncoder: Encoder[WellboreSource] = deriveEncoder
+  implicit val wellboreSourceItemsEncoder: Encoder[WellboreSourceItems] = deriveEncoder
+  implicit val wellboreItemsDecoder: Decoder[WellboreItems] = deriveDecoder
+  implicit val wellboreMergeRulesEncoder: Encoder[WellboreMergeRules] = deriveEncoder
+
+  implicit val emptyObjDecoder: Decoder[EmptyObj] = deriveDecoder
+  implicit val emptyObjEncoder: Encoder[EmptyObj] = deriveEncoder
+
+  implicit val jsonObjectItemsEncoder: Encoder[Items[JsonObject]] = deriveEncoder
+  implicit val jsonObjectItemsWithCursorDecoder: Decoder[ItemsWithCursor[JsonObject]] =
+    deriveDecoder
+
+  implicit val wellFilterEncoder: Encoder[WellFilter] = deriveEncoder
+  implicit val wellFilterRequestEncoder: Encoder[WellFilterRequest] = deriveEncoder
+
+  implicit val limitAndCursorEncoder: Encoder[LimitAndCursor] = deriveEncoder
+
+  private[v1] def discard[T](x: T): Unit = {
+    val _ = x
+  }
```

Comment:
what is this magic piece used for? 😮

Reply:
I added it because otherwise, it generated errors about unused variables. But there is probably a better way. I'll give it a second try :+1:

Reply:
Hmm, I could actually just replace it with
```scala
_ => ()
```